File: src/main/java/com/HOIIVUtils/hoi4utils/map/province/VictoryPoint.java
package com.HOIIVUtils.hoi4utils.map.province;

import com.HOIIVUtils.hoi4utils.clausewitz_data.Localizable;

/**
 * localizable: victory point name
 */
public class VictoryPoint implements Localizable {
	int value;
	int province;       // todo final?

	public VictoryPoint(int value, int province) {
		this.value = value;
		this.province = province;
	}

	public int value() {
		return value;
	}

	public void setValue(int value) {
		this.value = value;
	}

	public int province() {
		return province;
	}

	public void setProvince(int province) {
		this.province = province;
	}


}

File: src/main/scala/com/map/VictoryPoint.scala
package com.map

import com.hoi4utils.localization.{Localizable, Property}

import java.util
import scala.collection.mutable
import scala.collection.mutable.ListBuffer

/**
 * localizable: victory point name
 *
 * TODO: implement
 */
class VictoryPoint protected(var province: Int = 0, var value: Int = 0) extends Localizable:
  
  /**
   * Default method to get the localizable property identifiers and keys.
   *
   * @return a map of localizable property identifiers and keys.
   */
  override def getLocalizableProperties: mutable.Map[Property, String] = ???

  /**
   * Gets the localizable group of objects that this object is a part of.
   *
   * @return the localizable group of objects.
   */
  override def getLocalizableGroup: Iterable[? <: Localizable] = ???


object VictoryPoint:
  var victoryPoints: ListBuffer[VictoryPoint] = ListBuffer.empty
  var victoryPointErrors: ListBuffer[String] = ListBuffer.empty

